runtime/near-vm-runner/src/wasmer1_runner.rs

```diff
@@ -94,19 +96,58 @@ impl IntoVMError for wasmer::RuntimeError {
         match self.downcast::<VMLogicError>() {
-            Ok(e) => (&e).into(),
-            _ => {
-                if let Some(trap_code) = trap_code {
-                    // A trap
-                    VMError::FunctionCallError(FunctionCallError::Wasmer1Trap(
-                        trap_code.to_string(),
-                    ))
-                } else {
-                    // A general error
-                    VMError::FunctionCallError(FunctionCallError::WasmerRuntimeError(error_msg))
-                }
-            }
+            Ok(e) => return (&e).into(),
+            _ => {}
         }
+        let error = match trap_code {
+            Some(TrapCode::StackOverflow) => FunctionCallError::WasmTrap(WasmTrap::StackOverflow),
+            Some(TrapCode::HeapSetterOutOfBounds) => {
+                FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds)
+            }
+            Some(TrapCode::HeapAccessOutOfBounds) => {
+                FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds)
+            }
+            Some(TrapCode::HeapMisaligned) => {
+                FunctionCallError::WasmTrap(WasmTrap::MisalignedAtomicAccess)
+            }
+            Some(TrapCode::TableSetterOutOfBounds) => {
+                FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds)
+            }
+            Some(TrapCode::TableAccessOutOfBounds) => {
+                FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds)
+            }
+            Some(TrapCode::OutOfBounds) => FunctionCallError::WasmTrap(WasmTrap::MemoryOutOfBounds),
+            Some(TrapCode::IndirectCallToNull) => {
+                FunctionCallError::WasmTrap(WasmTrap::IncorrectCallIndirectSignature)
+            }
+            Some(TrapCode::BadSignature) => {
+                FunctionCallError::WasmTrap(WasmTrap::IncorrectCallIndirectSignature)
+            }
+            Some(TrapCode::IntegerOverflow) => {
+                FunctionCallError::WasmTrap(WasmTrap::IllegalArithmetic)
+            }
+            Some(TrapCode::IntegerDivisionByZero) => {
+                FunctionCallError::WasmTrap(WasmTrap::IllegalArithmetic)
+            }
+            Some(TrapCode::BadConversionToInteger) => {
+                FunctionCallError::WasmTrap(WasmTrap::IllegalArithmetic)
+            }
+            Some(TrapCode::UnreachableCodeReached) => {
+                // Not FunctionCallError::WasmTrap(WasmTrap::Unreachable) for Wasmer 0.x compatibility.
+                FunctionCallError::WasmUnknownError
+            }
+            Some(TrapCode::Interrupt) => {
+                FunctionCallError::Nondeterministic("Wasmer interrupt".to_string())
+            }
+            Some(TrapCode::UnalignedAtomic) => {
+                FunctionCallError::WasmTrap(WasmTrap::MisalignedAtomicAccess)
+            }
+            Some(TrapCode::VMOutOfMemory) => {
+                FunctionCallError::Nondeterministic("Wasmer out of memory".to_string())
+            }
```

Comment:
```suggestion
            Some(TrapCode::UnalignedAtomic) => {
                FunctionCallError::WasmTrap(WasmTrap::MisalignedAtomicAccess)
            }
            Some(TrapCode::Interrupt) => {
                FunctionCallError::Nondeterministic("Wasmer interrupt".to_string())
            }
            Some(TrapCode::VMOutOfMemory) => {
                FunctionCallError::Nondeterministic("Wasmer out of memory".to_string())
            }
```

nit: can group non-deterministic errors together.

Reply:
Done.